ml_garden/core/pipeline.py

```diff
@@ -146,6 +144,14 @@ def _validate_configuration(cls, config: dict[str, Any]) -> None:
                 ' "parameters" section in the config.'
             )
 
+    @classmethod
+    def _validate_task(cls, task: str) -> Task:
+        try:
+            return Task(task.lower())
```

Comment:
This is preference, but we usually favor enum _names_ over _values_ (as values are just arbitrary, see the fact that `Enum` has `auto()` for auto populating values).

This translates to looking up values from config files with `[]` instead of `()`.

However, if you maintain the usual `SOMETHING = "something"` pattern of `str, Enum`s and use `.lower()` to compare, then this comment is completely moot.

The only relevant thing is that users know what to write in their settings. Is it the values of enum members? Is it the names? Does it not matter?